walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,69 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
 
+/** 
+ * Query to find all the possible time slots for a requested meeting.
+ * If there are one or more possible time slots that works for both
+ * mandatory and optional attendees, return this.
+ * If not return only possible time slots that works for the mandatory attendees.
+ */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    long duration = request.getDuration();
+
+    List<Event> byStartTime = events.stream()
+        .sorted(Comparator.comparing(Event::getWhen, TimeRange.ORDER_BY_START))
+        .collect(Collectors.toList());
```

Comment:
I think this can be sorted directly using sort and the Comparator, rather than being converted to a stream first. Something like ```events.sort(TimeRange.ORDER_BY_START)``` or maybe ```Collections.sort(events, TimeRange.ORDER_BY_START)```.

Reply:
I see, thanks! 
Yeah, I think I can avoid the stream. I will just need to convert the Collection to a List, so that I can apply the sort.